packages/kbn-expandable-flyout/src/components/settings_menu.tsx

```diff
@@ -142,9 +161,9 @@ export const SettingsMenu: React.FC<SettingsMenuProps> = memo(
                   toolTipContent: FLYOUT_TYPE_PUSH_TOOLTIP,
                 },
               ]}
-              idSelected={flyoutTypeProps.type as string}
+              idSelected={flyoutType || 'overlay'}
```

Comment:
similar to my answer above, at the time of the redux store creation we do not have any `urlKey` values so we can't really populate the store. Only when the flyout gets added to the DOM do we add things to the store. Because the push vs overlay values are saved by id (by `urlKey` to be precise) I don't think it makes sense to try to be smart and populate the store early.
The default value is alway `overlay` so I feel like returning the value here if nothing is in the store is ok. 

Thoughst?

Reply:
this is now fixed in my [last commit](https://github.com/elastic/kibana/pull/192745/commits/8e091aa08a04e2ec00357417cee519f3aa5351c0) as I'm not storing values by id anymore, I can easily initialize the store to `overlay` which simplified the logic in a bunch of places